include/artifact-uploader.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2017 Prism Skylabs
+ */
+#ifndef CONNECT_SDK_ARTIFACT_UPLOADER_H
+#define CONNECT_SDK_ARTIFACT_UPLOADER_H
+
+#include "domain-types.h"
+#include "boost/shared_ptr.hpp"
+#include "public-util.h"
+
+namespace prism
+{
+namespace connect
+{
+
+class Client;
+// Usage example
+// namespace prc = prism::connect;
+// void configCallback(prc::Client& client)
+// {
+//     client.setLogFlags(prc::Client::LOG_INPUT | prc::Client::LOG_INPUT_JSON);
+//     client.setConnectionTimeoutMs(5000);
+//     client.setLowSpeed(5);
+// }
+//
+// prc::ArtifactUploader::Configuration uploaderConfig("<apiRoot>", "<apiToken>",
+//      "camOne", 32);
+// prc::ArtifactUploader uploader;
+// prc::Status status = uploader.init(uploaderConfig, configCallback);
```

Comment:
C++ does good job handling throwing in constructors and it is kind of standard thing for C++.
However, blocking code in constructor. Probably not something one would expect, agree.